### PR TITLE
STUD-3852 Handle Heartbeat Replies

### PR DESCRIPTION
**Pull request by austinarbor-wk**
# Description
Track when heartbeat responses are received from reaper. If we haven't received a response in a long time, reconnect the socket.

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: socket_transport.go [hunk 1]
```diff
@@ -93,6 +95,13 @@ func (st *socketTransport) writer() {
 				continue
 			}
 
+			lastBeat := time.UnixMilli(st.lastHeartbeat.Load())
+
+			if time.Now().Sub(lastBeat) > pongWait && !st.getIsReconnecting() {
+				st.reconnect <- struct{}{}
```
Comment (alexandrachakeres-wk) [inline socket_transport.go:102]: What's this line doing? Something with channels?The empty struct literal is confusing me.

Comment (alexandrachakeres-wk) [reply socket_transport.go:102]: Ah I think it's just something to trigger this bit of the `supervisor` method
```go
select
...
case <-st.reconnect:
```

**Review by alexandrachakeres-wk — DISMISSED** [review]
LGTM

File: socket_transport.go (hunk 1 above)
Comment (austinarbor-wk) [reply socket_transport.go:102]: yea it just puts an empty struct on the channel so the case `<- st.reconnect` will match. The struct itself is unused

**Comment by austinarbor-wk** [comment]
@alexandrachakeres-wk i just added a log message to print out when it is disconnecting due to timeout, can you re-approve? Thanks!

**Review by alexandrachakeres-wk — APPROVED** [review]
+1

**Comment by austinarbor-wk** [comment]
QA +1

**Comment by austinarbor-wk** [comment]
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**